libs/gamify/auth/src/lib/guards/roles.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { Reflector } from "@nestjs/core";
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+
+    constructor(private reflector: Reflector) {}
+
+    canActivate(context: ExecutionContext): boolean {
+        const roles = this.reflector.get<string[]>('roles', context.getHandler());
```

Comment:
Future improvement: introduce a `MetaKeys` enum for this (or something similar).